resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/watson/ElementTree.java

```diff
@@ -206,7 +206,7 @@ public synchronized void createElement(IPath key, Object data) {
 		try {
 			tree.createChild(parent, key.lastSegment(), data);
 		} catch (ObjectNotFoundException e) {
-			elementNotFound(parent);
+			throw createElementNotFoundException(parent);
```

Comment:
I wonder why don't we supply original exception as root cause?

Reply:
as it never happens anway?